### DISCUSSION TITLE
Process to identify conflicting shortcuts

### DISCUSSION THREAD

**Discussion by sccardais** [Anze/KeyCluCask#102] [Ideas]
### Proposal

Shortcuts are terrific until they start doing something you don't expect. This can happen when different applications use the same shortcut. Any tool that helped to identify conflicting shortcuts would be very helpful.

### Alternatives

I've wondered if a tool like KeyClu might allow copying the shortcuts for a given application to a spreadsheet. This might allow the user to create a large table of shortcuts for different applications to help identify duplicates. Not sure it that will work in practice but it's something I've wondered about.

**Comment by Anze** [comment]
Hey [sccardais](https://github.com/sccardais),
Thanks for sharing the idea! If I understand correctly, you're looking for a way to list the shortcuts used by the app. If that's the case, there's a feature for that: after enabling "Show command bar" in `Settings` → `Window`, you can click the `M` button on the bottom right Shortcuts Panel to export the app's shortcuts to a markdown file. While it's not a spreadsheet, it still provides a good overview. Let me know if this works for you, or if you're looking for something different.

**Comment by sccardais** [comment]
Thank you. I’ll let you know if it works as I’m hoping. Essentially, I’m
hoping to create a database of shortcuts for my applications. Initially, it
will have to be the default shortcuts but it’s a start.

**Comment by FFSFFS** [comment]
Hi,
First, congratulations on the great app!
I'm looking for a way to identify which app is using a specific shortcut. I'm experiencing shortcut overlap in Sequoia and I'm not sure how to resolve it. Could Keyclu help with this?

Thanks a lot!

**Comment by Anze** [comment]
Hey [FFSFFS](https://github.com/FFSFFS),
The feature you’re asking for isn’t something app can handle right now. However, there’s a [ShortcutDetective](https://www.irradiatedsoftware.com/labs/) app that does exactly what you need. Check it out – it should have you covered!

**Comment by ffs-unifesp** [comment]
Hi, @Anze. Thanks for the prompt reply.

It turns out that ShortcutDetective is not working on sequoia (intel and m3). It keeps crashing whenever any shortcut is pressed. 

any help?

**Comment by Anze** [comment]
Hey [ffs-unifesp](https://github.com/ffs-unifesp),

It’s unfortunate that ShortcutDetective is crashing on newer macOS versions. I’m still using macOS 13, and while app not completely stable, it does work. I’ll keep your request in mind as a potential feature for future updates.

**Reply by CJSen** [reply]
I hope this feature can be launched soon！Thank you～！

**Comment by mahmoudimus** [comment]
Hi @ffs-unifesp - I fixed the crash that happens with `ShortcutDetective` if you're interested.

**Reply by CJSen** [reply]
> Hi @ffs-unifesp - I fixed the crash that happens with `ShortcutDetective` if you're interested.嗨-我修复了 `ShortcutDetective` 发生的崩溃，如果你感兴趣的话。

hi，bro。How to get your repair method?Thank you！

**Reply by amoshkov** [reply]
> Hi @ffs-unifesp - I fixed the crash that happens with `ShortcutDetective` if you're interested.

Hey @mahmoudimus, can you share more details? I need Shortcut Detective to find one annoying shortcut that I want to kill. Thanks!

**Reply by ffs-unifesp** [reply]
where are the fix?
